compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -882,14 +882,16 @@ static LogicalResult setMatmulNoPadRootConfig(
     bool allowIncompleteTile =
         vecPreProcStrategy == VectorPreProcStrategy::Peeling ||
         vecPreProcStrategy == VectorPreProcStrategy::Masking;
+    // The backend struggles to legalize non-power-of-two scalable vectors.
```

Comment:
[optional] I wonder if we have an issue for tracking this? Will we address the issue for non-power-of-two cases?

Reply:
I don't think it's something we can directly address, but it is a known issue in the backend, that we could track. cc @banach-space.

Reply:
Non-powers of 2 are tricky because scalarization of scalable vectors is not as straightforward as for fixed-width vectors.

These things fall into the category of "we know that this doesn't work great, but also nobody is complaining". In this case we can easily tweak IREE without any performance penalties. With a rather non-intrusive patch. So I'd leave it as a "TODO". I hope that this makes sense :)

@MacDue Perhaps add a note/todo that relaxing this condition would require investigating (and potentially updating) the middlend/backend. Btw, AFAIK, this will apply to scalable vectors in general rather than just SVE.

Reply:
I see, thanks thanks! No action item is required, I was just curious. :)

Reply:
Let me land this then :)